### PR TITLE
Dscp marking and annotaion for qospolicy

### PR DESCRIPTION
**Pull request by niteshcha**

### CODE REVIEW

File: pkg/qospolicy/apis/aci.qos/v1/types.go [hunk 1]
```diff
@@ -22,12 +22,14 @@ type QosPolicySpec struct {
 	Selector PodSelector  `json:"selector,omitempty"`
 	Ingress  PolicingType `json:"ingress,omitempty"`
 	Egress   PolicingType `json:"egress,omitempty"`
+	DscpMark PolicingType `json:"dscpmark,omitempty"`
```
Comment (jayaramsatya) [inline pkg/qospolicy/apis/aci.qos/v1/types.go:25]: why did you define DscpMark here as PolicingType.  can you post your sample CR yaml file here

Comment (niteshcha) [reply pkg/qospolicy/apis/aci.qos/v1/types.go:25]: apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: qospolicies.aci.qos
spec:
  group: aci.qos
  version: v1
  names:
    kind: QosPolicy
    listKind: QosPolicyList
    plural: qospolicies
    singular: qospolicy
  scope: Namespaced
  validation:
    openAPIV3Schema:
      properties:
        apiVersion:
          type: string
        kind:
          type: string
        spec:
          properties:
            podSelector:
              description: 'Selection of Pods'
              properties:
                matchLabels:
                  type: object
                  description:
            ingress:
              properties:
                policing_rate:
                  type: integer
                policing_burst:
                  type: integer
            egress:
              properties:
                policing_rate:
                  type: integer
                  minimum: 0
                policing_burst:
                  type: integer
                  minimum: 0
            dscpmark:
              properties:
                dscp_marking:
                  type: integer
                  minimum: 0
                  maximum: 56
---

Comment (niteshcha) [reply pkg/qospolicy/apis/aci.qos/v1/types.go:25]: i have tested creating a container with above sample file and then created a policy using
apiVersion: aci.qos/v1
kind: QosPolicy
metadata:
  name: qos-policy1
  namespace: default
spec:
  podSelector:
    matchLabels:
      role: db
  ingress:
    policing_rate : 1000
    policing_burst: 1000
  egress:
    policing_rate:  1000
    policing_burst: 1000
  dscpmark:
    dscp_marking:   18
~                          
i can see mo getting pushed and populated as expected

Comment (jayaramsatya) [reply pkg/qospolicy/apis/aci.qos/v1/types.go:25]: Based on your config, DscpMark can be just Integer type right ? why do you need type to be PolicingType. and also remove the dscpmark from PolicingType.

Comment (niteshcha) [reply pkg/qospolicy/apis/aci.qos/v1/types.go]: Done

**Review by jayaramsatya — APPROVED** [review]
LGTM

**Pull request merged**